go/tasks/plugins/array/k8s/task.go

```diff
@@ -82,6 +104,7 @@ func (t Task) Launch(ctx context.Context, tCtx core.TaskExecutionContext, kubeCl
 	pod = ApplyPodPolicies(ctx, t.Config, pod)
 	pod = applyNodeSelectorLabels(ctx, t.Config, pod)
 	pod = applyPodTolerations(ctx, t.Config, pod)
+	pod = addPodFinalizer(pod)
```

Comment:
We should gate this on the same or similar config: https://github.com/lyft/flyteplugins/blob/master/go/tasks/pluginmachinery/flytek8s/config/config.go#L53